api/stats.js

```diff
@@ -0,0 +1,101 @@
+import { config } from './../config.js';
+
+export class Stats {
+    constructor(discordClient) {
+        this.client = discordClient;
+        this.stats = {};
+    }
+
+    initializeStats(prev_message) {
+        /*
+        This is a placeholder.
+        We have some limitations on how many messages we can load per request
+        we can have daily pulls, use the latest message as the last source of truth
+        and update with new stats
+        let stats = {
+            username1: {
+                messages: 5, 
+                words: 20, 
+                letters: 150
+            }
+        }
+       */
+        return {};
+    }
+
+    async _buildStats(channels, after) {
+        let stats = initializeStats();
+
+        for (let [_, channel] of channels) {
+            if (channel.type !== 'GUILD_TEXT') {
+                continue;
+            }
+            const messages = await channel.messages.fetch({ limit: 100, after: after })
+            for (let [_, message] of messages) {
+                let username = message.author.username;
+                let user_stats = stats[username] = stats[username] || {
+                    posts: 0,
+                    words: 0,
+                    letters: 0
+                };
+                user_stats.posts++;
+                user_stats.words += message.content.split(' ').length;
+                user_stats.letters += message.content.length;
+            }
+        }
+
+
+        return stats;
+    }
+
+    async _buildStats(channels, after) {
```

Comment:
isn't this function duplicated?

Reply:
lol, it is, that `rebase` got me last night